lib/cfg/cfg_Node.mli

```diff
@@ -0,0 +1,54 @@
+(** The module provides the abstraction of a CFG node. *)
+module Node : sig
+  type 'a t [@@deriving show, sexp]
+
+  (** Checks if two nodes and their children are the same recursively.
+
+      Notes: Should always work for the structure generated by {{! Cfg.Converter}Converter}.
+      Severely limited, only checks isomorphism for two children for each node,
```

Comment:
Why there are these limitations? What are the possible problems we should be aware of?

Reply:
And why we have to compare two nodes?

Reply:
The limitations are because graph ismorphism is not feasable, the problems are simply that it may not return true if the two nodes are actually the same.
We need to compare two nodes for the unit tests, otherwise its not really needed.